schema/resource.json

```diff
@@ -16,12 +16,7 @@
         "Attributes": {
             "title": "Attributes",
             "type": "object",
-            "additionalProperties": true,
-            "properties": {
-                "service.name": {
-                    "type": "string"
-                }
-            }
+            "additionalProperties": true
```

Comment:
Just so I understand - does the go implementation struggle with all places where we set `additionalProperties: true`? I think being able to read additional properties will be critical to supporting custom extension components.

Reply:
correct... i will try to submit a fix for supporting additional properties

Reply:
@jack-berg to be clear I'm not suggesting that we remove support for additional properties. Running into this issue in go-jsonschema made me wonder if attributes needed any properties defined or not, this is why I opened a PR to remove the service name

Reply:
Gotcha. Yeah the "required" feels wrong to me. While an SDK is required to produce a resource with `service.name`, there is no requirement that it is specified by `OTEL_SERVICE_NAME` or the user, since there is a well defined fallback value.

I think the same should probably apply here: drop the `service.name` requirement, but heavily encourage in the example documentation.